### ISSUE TITLE
When I try to store a LocalDate, it crashes.

### ISSUE THREAD

**Issue by AlexMakesSoftware** [AlexMakesSoftware/GSON_demo#1]
Error: 
Exception in thread "main" com.google.gson.JsonIOException: Failed making field 'java.time.LocalDate#year' accessible; either change its visibility or write a custom TypeAdapter for its declaring type
...
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field private final int java.time.LocalDate.year accessible: module java.base does not "opens java.time" to unnamed module @3fd7a715

I don't quite understand but it has something to do with the new module system in Java, see: See: http://www.javawenti.com/?post=4674
Basically I don't think you have permission to do what the reflection code is asking to do in the GSON api.

TODO: investigate this 'TypeAdapter'.

**Comment by AlexMakesSoftware** [comment]
TypeAdapter is the way to go and it's pretty easy really.

See: https://www.tutorialspoint.com/gson/gson_custom_adapters.htm
